Do not automatically reset the shard duration when using ALTER RETENTION POLICY

jsternberg opened the pull request:
Fixes #7150.

jsternberg commented on services/meta/errors.go:
```diff
@@ -55,6 +55,10 @@ var (
 
+	// ErrRetentionPolicyDurationBelowShardDuration is returned when updating a retention
+	// policy that has a duration lower than the current shard duration.
+	ErrRetentionPolicyDurationBelowShardDuration = errors.New("retention policy duration must be greater than the shard duration")
```
Hm, it is a really long name, but I think it matches with the naming convention of the other errors. I don't think `ErrShortDuration` says enough about the error since it can also applies to `ErrRetentionPolicyDurationTooLow`.

joelegasse replied:
`ErrIncompatibleDurations`?

jsternberg replied:
I can do that. But, I'm going to do `ErrRetentionPolicy` as the prefix since that's the naming standard. I know it's long, but I don't want to break that naming. So is `ErrRetentionPolicyIncompatibleDurations` fine?

joelegasse replied:
I think that's excessive... `ErrReplicationFactorTooLow` isn't prefixed with `RetentionPolicy`...

I'll let other people weigh in on the excessively long names

e-dard replied:
@jsternberg how about changing `ErrRetentionPolicyDurationTooLow` to `ErrRetentionPolicyDurationInvalid`, and then instead of `ErrRetentionPolicyDurationBelowShardDuration` have `ErrRetentionPolicyDurationTooLow` ?

jsternberg replied:
I changed it to what @joelegasse suggested originally.

jsternberg commented on services/meta/client_test.go:
```diff
@@ -360,6 +360,77 @@ func TestMetaClient_SetDefaultRetentionPolicy(t *testing.T) {
+		t.Fatalf("expected error '%s', got '%s'", meta.ErrRetentionPolicyDurationBelowShardDuration, err)
+	}
+
+	// Set the shard duration above the shard group duration. This should also return an error.
```
Whoops. It seems I messed up changing the second half of this comment. Fixed.

jsternberg commented:
Updated. I also found that it didn't work when the duration was 0 (infinite time) and it didn't prevent creates so I added test cases for both of those and fixed it.
